src/blk/kernel/KernelDevice.cc

```diff
@@ -808,11 +808,12 @@ void KernelDevice::_discard_thread(uint64_t tid)
       // It will also allow threads to finish in a timely manner.
       constexpr unsigned MAX_LOCAL_DISCARD = 32;
       unsigned count = 0;
-      for (auto p = discard_queued.begin();
-	   p != discard_queued.end() && count < MAX_LOCAL_DISCARD;
-	   ++p, ++count) {
-	discard_processing.insert(p.get_start(), p.get_len());
-	discard_queued.erase(p);
+      auto it = discard_queued.begin();
+      while (it != discard_queued.end() && count < MAX_LOCAL_DISCARD) {
+        auto current = it++;  // Save current and move iterator forward
```

Comment:
erase() returns iterator following the removed entry. So one can use:
it = erase(it);

Reply:
it = discard_queued.begin();
should be fine here as well.

Reply:
> erase() returns iterator following the removed entry. So one can use: it = erase(it);

erase() of `interval_set` class returns void, not a iterator. : (

Reply:
@YiteGu 
Maybe we should modify `interval_set::erase()` ?

Reply:
> @YiteGu Maybe we should modify `interval_set::erase()` ?

yeap, I try to modify it now.

Reply:
> @YiteGu Maybe we should modify `interval_set::erase()` ?

done